spec/API_specification/statistical_functions.md

```diff
@@ -113,7 +126,7 @@ Calculates the product of input array `x` elements.
 
 -   **out**: _&lt;array&gt;_
 
-    -   if the product was computed over the entire array, a zero-dimensional array containing the product; otherwise, a non-zero-dimensional array containing the products. The returned array must have the same data type as `x`.
+    -   if the product was computed over the entire array, a zero-dimensional array containing the product; otherwise, a non-zero-dimensional array containing the products.
```

Comment:
How about
```suggestion
    -   if the product was computed over the entire array, a zero-dimensional array containing the product; otherwise, a non-zero-dimensional array containing the products. The returned array has a data type as described by the `dtype` parameter above.
```